### PR TITLE
[ADD] OCA codesprint fixes

### PR DESCRIPTION
**Pull request by hbrunn**
- a few bugfixes, including the grave one that everyone read with admin
  permissions
- write/create attachment support

### CODE REVIEW

File: document_sftp/document_sftp_server.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def check_auth_password(self, username, password):
             if not user:
                 return AUTH_FAILED
             user.sudo(user.id).check_credentials(password)
+            self.env = self.env(user=user.id)
```
Comment (pedrobaeza) [inline document_sftp/document_sftp_server.py:22]: Why not `self.sudo(user)`?

File: document_sftp/document_sftp_server.py [hunk 2]
```diff
@@ -37,6 +38,7 @@ def check_auth_publickey(self, username, key):
                     'Ignoring key of unknown type for line %s', line)
                 continue
             if RSAKey(data=decodebytes(key_data)) == key:
+                self.env = self.env(user=user.id)
```
Comment (pedrobaeza) [inline document_sftp/document_sftp_server.py:41]: The same as above

File: document_sftp/models/document_sftp_root.py [hunk 3]
```diff
@@ -59,6 +63,18 @@ def _open(self, path, flags, attr):
         raise NotImplementedError()
 
     @api.model
+    def _remove(self, path):
+        """Return file attributes"""
+        raise NotImplementedError()
```
Comment (pedrobaeza) [inline document_sftp/models/document_sftp_root.py:68]: Is it recommended to return an error at this level? Shouldn't we return a forbidden code or similar?

File: document_sftp/models/document_sftp_root.py [hunk 4]
```diff
@@ -59,6 +63,18 @@ def _open(self, path, flags, attr):
...
+    @api.model
     def _lstat(self, path):
         """Return attributes about a link"""
         return self._stat(path)
+
+    @api.model
+    def _split_path(self, path):
+        """Return a list of normalized and stripped path components"""
+        # TODO: normalization
+        path = path.strip('/')
```
Comment (pedrobaeza) [inline document_sftp/models/document_sftp_root.py:79]: Shouldn't you use path separator from `os.path`?

File: document_sftp/document_sftp_server.py (hunk 2 above)
Comment (hbrunn) [reply document_sftp/document_sftp_server.py:41]: note that the files which are located in the module's root (so not in `models`) are not models, but implementations of the respective interfaces that need to be implemented for paramiko

File: document_sftp/models/document_sftp_root.py (hunk 4 above)
Comment (hbrunn) [reply document_sftp/models/document_sftp_root.py:79]: no, the ssh protocol defines the path separator as bein a slash: https://tools.ietf.org/id/draft-ietf-secsh-filexfer-13.txt (section 6, file names)

File: document_sftp/models/document_sftp_root.py (hunk 3 above)
Comment (hbrunn) [reply document_sftp/models/document_sftp_root.py:68]: good question. I don't really have an opinion about this, the good thing about the exception is that it's quite clear what's going on, whereas a standard return value might also point to a permission issue or the like

File: document_sftp/document_sftp_server.py (hunk 2 above)
Comment (pedrobaeza) [reply document_sftp/document_sftp_server.py:41]: OK, thanks for the explanation

File: document_sftp/models/document_sftp_root.py (hunk 3 above)
Comment (pedrobaeza) [reply document_sftp/models/document_sftp_root.py:68]: OK, we keep as is right now.

**Comment by pedrobaeza** [comment]
Can you please review Travis status?

**Comment by hbrunn** [comment]
this module never went to production, and should be redesigned (ie own workers instead of threads) if ever this was picked up again

**Pull request closed without merging**